docs/source/dataframe-parquet.rst

```diff
@@ -0,0 +1,234 @@
+filesystem-specific configuration files/environment variables (e.g. the
+``~/.aws/config`` file for S3). Alternatively, you can pass configuration on to
+the fsspec_ backend through the ``storage_options`` kwarg:
+
+.. code-block:: python
+
+   >>> df = dd.read_parquet(
+   ...      "s3://bucket-name/my/parquet/",
+   ...      storage_options={'anon': True}  # passed to `s3fs.S3FileSystem`
+   ... )
+
+`read_parquet` has many configuration options affecting both behavior and
+performance. Here we highlight a few common options.
+
+Engine
+~~~~~~
+
+`read_parquet` supports two backend engines - ``fastparquet`` and ``pyarrow``.
+For historical reasons this defaults to ``fastparquet`` (if installed), and
+falls back to ``pyarrow``. We recommend using ``pyarrow`` when possible. This
+can be explicitly set by passing ``engine="pyarrow"``, but if ``fastparquet``
+is not installed this configuration is unnecessary since it will automatically
+fallback to ``pyarrow``.
+
+.. code-block:: python
+
+   >>> df = dd.read_parquet(
+   ...      "s3://bucket-name/my/parquet/",
+   ...      engine="pyarrow"  # explicitly specify the pyarrow engine
+   ... )
+
+Metadata
+~~~~~~~~
+
+When `read_parquet` is called, it first loads metadata about the file(s). This
+metadata may include:
+
+- The dataset schema
+- How the dataset is partitioned into files (and those files into row-groups)
+
+Some parquet datasets include a ``_metadata`` file which aggregates per-file
+metadata into a single location. For small-to-medium sized datasets this *may*
+be useful - it makes accessing the row-group metadata possible without reading
+parts of *every* file in the dataset. Row-group metadata allows dask to split
+large files into smaller in-memory partitions, and merge many small files into
+larger partitions, possibly leading to higher performance.
+
+However, for large datasets the ``_metadata`` file can be problematic - it may
+be too large for a single endpoint to parse! If this is true, you can disable
+loading the ``_metadata`` file by specifying ``ignore_metadata_file=True``.
+
+.. code-block:: python
+
+   >>> df = dd.read_parquet(
+   ...      "s3://bucket-name/my/parquet/",
+   ...      ignore_metadata_file=True  # don't read the _metadata file
+   ... )
+
+If no ``_metadata`` file is present, dask will load each parquet file
+individually as a partition in the larger dask dataframe (unless only a single
+file is specified). This is performant provided all files are of reasonable
+size (We recommend aiming for 10-250 MiB in-memory size per file once loaded
+into pandas). Too large files can lead to excessive memory usage on a single
+worker, while too small files can lead to poor performance as the overhead of
+dask dominates. If you need to read a parquet dataset composed of many large
+files (and lacking a ``_metadata`` file), you can pass
+``gather_statistics=True`` to have dask load the row-group metadata from all
```

Comment:
I like that you are doing your best to keep the discussion simple, but my preference moving forward may be to deprecate `gather_statistics` as a public option. Right now, the only time the user should ever specify `gather_statistics`, is when they are doing it to insist on `gather_statistics=False`.  For the case you are highlighting here, the suggested argument is `split_row_groups=` (which doesn't actually require any row-group statistics to be gathered, but does require the parquet metadata to be parsed, as you said).

To generalize, there are really ~3 different flavors of `read_parquet`:

1. Simply read in distinct parquet files as Dask-DataFrame partitions, without calculating divisions, filtering data, or encoding directory partitions as columns. This takes the least amount of time on the client, because the Dask graph is extremely simple to define.  We will not need to parse metadata or gather statistics at all. As you stated above, this is the default when there is no `_metadata` file present (or when it is ignored).
2. Map an explicit number of row-groups in the dataset into Dask-DataFrame partitions and/or read in a directory-partitioned dataset, without calculating divisions, filtering data, or aggregating files. This requires us to parse the footer metadata, which is why it is not the default behavior when there is no `_metadata` file.  However, this does **not** require us to gather statistics, and so the suggested way to get this behavior is by setting `split_row_groups=True` (or to set an integer number of row-groups).
3. All other functionality (i.e. aggregating files, calculating divisions, and/or filtering row-groups) requires us to parse metadata **and** gather statistics.

Given this breakdown, it seems the most users will only need to gather statistics if they are hoping to calculate divisions or filter data. If neither of these cases are true, they only need to parse metadata if they want `split_row_groups=True`.  This is why I am wondering if it would make sense to throw out `gather_statistics`, and replace it with an argument that is more focused on the `divisions` question.

Reply:
Ah, thanks Rick. I was confused here and thought `gather_statistics` was a switch giving permission to load the footer from every file, not something more statistics-focused. I agree that we should maybe rethink these options to better serve the different ways a user might combine them. I'll open an issue where we can discuss this further.

For now, switching to highlighting `split_row_groups` here.

Reply:
https://github.com/dask/dask/issues/8937